File: examples/python-lifecycle/lifecycle.py
"""Full ERC-8183 lifecycle on Arc testnet — single-wallet self-roleplay.

Run: `python lifecycle.py` after setting PRIVATE_KEY in .env.
"""

import os
import sys

from dotenv import load_dotenv

from arckit import ArcKit, JobStatus

load_dotenv()

private_key = os.environ.get("PRIVATE_KEY")
if not private_key or private_key == "0x":
    sys.exit("Set PRIVATE_KEY in .env (and fund it via the Arc faucet)")

arc = ArcKit(
    private_key=private_key,
    network="testnet",
    rpc_url=os.environ.get("RPC_URL"),
)
me = arc.address
print(f"Wallet: {me}")

balance = arc.usdc.balance_of(me)
print(f"USDC balance: {balance}")

if balance < 0.01:
    sys.exit("Address has < 0.01 USDC. Fund it from the Arc faucet first.")

# Single wallet plays client + provider + evaluator for a self-contained demo.
print("\n[1/4] Creating job...")
job_id = arc.commerce.create_job(
    provider=me,
    evaluator=me,
    description="demo: self-roleplayed code review",
    expired_at_hours=1,
)
print(f"  Job created: id={job_id}")

print("\n[2/4] Setting budget (0.01 USDC)...")
arc.commerce.set_budget(job_id, 0.01)
print("  Budget set.")

print("\n[3/4] Funding job...")
arc.commerce.fund(job_id)  # auto-approves USDC
print("  Job funded.")

print("\n[4/4] Submitting deliverable + completing...")
arc.commerce.submit(job_id, "demo deliverable hash")
arc.commerce.complete(job_id, "looks good")

final = arc.commerce.get_job(job_id)
print(f"\nFinal status: {JobStatus(final.status).name} ({final.status})")
print("Explorer: https://testnet.arcscan.app")

File: packages/create-arc-app/templates/default/src/lifecycle.ts
import { ArcKit, JobStatus } from '@arckit/sdk';
import { privateKeyToAccount } from 'viem/accounts';

const PRIVATE_KEY = process.env.PRIVATE_KEY;
if (!PRIVATE_KEY || PRIVATE_KEY === '0x') {
  throw new Error('Set PRIVATE_KEY in .env (and fund the address via the Arc faucet)');
}

const account = privateKeyToAccount(PRIVATE_KEY as `0x${string}`);
const arc = new ArcKit({ account, network: 'testnet', rpcUrl: process.env.RPC_URL });

console.log(`Wallet: ${account.address}`);

const usdcBalance = await arc.usdc.balanceOf(account.address);
console.log(`USDC balance: ${usdcBalance}`);

if (usdcBalance < 0.01) {
  throw new Error(
    'Address has < 0.01 USDC. Fund it from the Arc faucet at https://testnet.arcscan.app first.',
  );
}

// For a self-contained demo we use the same wallet as client, provider, and evaluator.
// In a real app these would be three different addresses.
console.log('\n[1/4] Creating job...');
const jobId = await arc.commerce.createJob({
  provider: account.address,
  evaluator: account.address,
  expiredAt: Math.floor(Date.now() / 1000) + 3600,
  description: 'demo: self-roleplayed code review',
});
console.log(`  Job created: id=${jobId}`);

console.log('\n[2/4] Setting budget (0.01 USDC)...');
await arc.commerce.setBudget(jobId, 0.01);
console.log('  Budget set.');

console.log('\n[3/4] Funding job...');
await arc.commerce.fund(jobId); // auto-approves USDC
console.log('  Job funded.');

console.log('\n[4/4] Submitting deliverable + completing...');
await arc.commerce.submit(jobId, 'demo deliverable hash');
await arc.commerce.complete(jobId, 'looks good');

const final = await arc.commerce.getJob(jobId);
console.log(`\nFinal status: ${JobStatus[final.status]} (${final.status})`);
console.log('Explorer: https://testnet.arcscan.app/tx');
